src/main/java/seedu/application/logic/commands/InterviewEditCommand.java

```diff
@@ -0,0 +1,213 @@
+import static seedu.application.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.application.logic.parser.CliSyntax.*;
+import static seedu.application.logic.parser.CliSyntax.PREFIX_INTERVIEW_ADDRESS;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.application.commons.core.index.Index;
+import seedu.application.commons.util.CollectionUtil;
+import seedu.application.commons.util.ToStringBuilder;
+import seedu.application.logic.Messages;
+import seedu.application.logic.commands.exceptions.CommandException;
+import seedu.application.model.Model;
+import seedu.application.model.job.Job;
+import seedu.application.model.job.interview.Interview;
+import seedu.application.model.job.interview.InterviewAddress;
+import seedu.application.model.job.interview.InterviewDateTime;
+import seedu.application.model.job.interview.InterviewType;
+
+/**
+ * Edits an interview to a job in the application book.
+ */
+
+public class InterviewEditCommand extends InterviewCommand {
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = InterviewCommand.COMMAND_WORD + " " + COMMAND_WORD
+            + ": Edits an interview of a job. \n"
+            + "Parameters: INDEX (of interview) "
+            + PREFIX_JOB_SOURCE + "INDEX (of job)"
+            + "[" + PREFIX_INTERVIEW_TYPE + "INTERVIEW TYPE]"
+            + "[" + PREFIX_INTERVIEW_DATETIME + "INTERVIEW DATE AND TIME]"
+            + "[" + PREFIX_INTERVIEW_ADDRESS + "INTERVIEW ADDRESS]\n"
+            + "Example: \n"
+            + "1 " + PREFIX_JOB_SOURCE + "2 "
+            + PREFIX_INTERVIEW_TYPE + "Technical "
+            + PREFIX_INTERVIEW_DATETIME + "Dec 31 2030 1200 "
+            + PREFIX_INTERVIEW_ADDRESS + "Home\n"
+            + "edits 1st interview from 2nd job";;
+
+    public static final String MESSAGE_SUCCESS = "Interview successfully edited: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided. \n"
+            + PREFIX_INTERVIEW_TYPE + " for Interview Type\n"
+            + PREFIX_INTERVIEW_DATETIME + " for Interview Date and Time\n"
+            + PREFIX_INTERVIEW_ADDRESS + " for Interview Address\n";
+    public static final String MESSAGE_DUPLICATE_INTERVIEW = "This interview already exists for the job";
+
+    private final Index jobIndex;
+    private final Index interviewIndex;
+    private final EditInterviewDescriptor editInterviewDescriptor;
+
+    /**
+     * @param jobIndex index of the job in the filtered job list to edit
+     * @param interviewIndex index of the interview in the job to edit.
+     * @param editInterviewDescriptor details to edit the interview with
+     */
+    public InterviewEditCommand(Index jobIndex, Index interviewIndex, EditInterviewDescriptor editInterviewDescriptor) {
+        requireAllNonNull(jobIndex, interviewIndex);
+        requireNonNull(editInterviewDescriptor);
```

Comment:
could try to combine into one line, requireAllNonNull(jobIndex, interviewIndex, editInterviewDescriptor)